docs/paylinks_and_payment_request.md

```diff
@@ -0,0 +1,87 @@
+
+## High Level Algorithm
+
+1. Given a `<link rel="facilitated-payment" href="<scheme>://<host>/<path>?<query>">`
+2. If the `<scheme>` is "https", then:
+    1. Construct a PaymentRequest object with parameters:
+        - `supportedInstruments: "<scheme>://<host>/<path>" (e.g., "https://wallet.example/pay")`
+        - `data: ConvertQueryToDictionary(<query>)` (to be defined)
+    2. The user agent checks [PaymentRequest.canMakePayment()](https://w3c.github.io/payment-request/#canmakepayment-method) for the constructed Payment Request
+    3. If true, the user agent displays some UX that lets user trigger the payment handler.
+    4. If the user selects to trigger this payment handler, trigger [PaymentRequest.show()](https://w3c.github.io/payment-request/#show-method).
+    5. The user agent would then drop the response from PaymentRequest.show(). (Payment links are designed for merchants that have back-end integration with payment apps, where the back-end (server-to-server) calls are responsible for communication of a successful transaction.)
+3. Else if the `<scheme>` is not "https", then see [How to handle non-HTTPS schemes?](#how-to-handle-non-https-schemes) below!
+
+## Open Questions
+
+### When would the UX be shown to the user to offer them the payment handler?
+
+Firstly, it is likely that we would specify (normatively or non-normatively)
+that the browser wait until document load has settled before triggering any UX
+for payment links on the page. We would likely also specify a short pause
+*after* a payment link is newly added to the page, to allow for collecting
+multiple payment links together (see [Handling multiple paylinks on the
+page](#handling-multiple-paylinks-on-the-page)).
+
+In addition, it is important for user agents to make sure that payment links do
+not become a vector for spamming users with un-wanted browser UX. This is also
+a problem for Payment Request, and is why the spec [allows browsers to require
+a transient user activation](https://w3c.github.io/payment-request/#show-method)
+to call `show()`, and generally allows the user agent to reject `show()` calls
+if it believes they are not legitimate.
+
+### Handling multiple paylinks on the page
+
+A site may want to include multiple payment links on the same page (perhaps
+coming from integrations with different PSPs or payment method libraries). This
+can likely be treated similarly to a single PaymentRequest call with multiple
+[PaymentMethodData](https://w3c.github.io/payment-request/#dom-paymentmethoddata)),
+if we specify some logic around 'collecting together' multiple payment links on
+the page.
+
+### Is it useful for pages to be able to have an event listener for the outcome, or a way to hook into the Payment Request that is triggered?
+
+This is definitely feasible (e.g. via defining `onpaymentrequestshow` or
+`onpaymentrequestcomplete`, etc), and simply needs us to determine if its useful.
+
+### How to handle non-HTTPS schemes?
```

Comment:
Also maybe something about "How can a payment app enable their payment URLs without writing separate code for each browser?" https://w3c.github.io/payment-handler/ and https://web.dev/articles/android-payment-apps-developers-guide answer this for payment websites and Android apps, but there's a gap for desktop browsers, especially when the wallet is stored on a different device. Would a [standard](https://www.w3.org/community/webextensions/) extension API be a full answer for that?

Reply:
cc @RByers  @ianbjacobs 

To me, this question is broader than paylinks and doesn't really fit here? Instead, it's really a question for Payment Request/Payment Handler itself; we don't have a specification or concept today of "more-native" apps on Desktop (or "cross-device" apps in general).

I would agree that it's probably going to be some form of extension API, in the absence of Desktop OS makers providing access to native apps on their platforms. Although that then begs the question of - why is a web app not good enough but an extension app would be? :)

Reply:
Yes today the only PaymentRequest extensibility story on desktop is PaymentHandler and IMHO that is fine, we haven't heard much need for native desktop wallet apps. But certainly a feature request to enable browser-extension-based PaymentHandlers is a totally reasonable request to consider. I agree with Stephen that it's orthogonal to this document.